log4j-layout-template-json/src/main/java/org/apache/logging/log4j/layout/template/json/util/JsonWriter.java

```diff
@@ -572,6 +565,10 @@ public void writeString(
 
     }
 
+    public void writeAsString(Object object) {
```

Comment:
This could be a `writeString` overload, but I thought it would be easier to understand that it takes potentially non-string values and formats them as strings with a slightly different name. I could go either way on this.

Reply:
This public method allows one to call `jsonWriter.writeAsString("here I break the JSON: \"")`.
1. I don't think this should be public. I would rather go with the old approach, where `ParameterizedMessage.deepToString()` was explicit. This method doesn't provide, IMHO, anything, except saving us to type a couple more characters.
2. All public methods in `JsonWriter` have sufficient user input protection, e.g., null checks, quoting of literals; `writeAsString()` doesn't.

Reply:
> I don't think this should be public. I would rather go with the old approach, where ParameterizedMessage.deepToString() was explicit. This method doesn't provide, IMHO, anything, except saving us to type a couple more characters.

The issue I ran into was using deepToString efficiently requires a (static, to avoid garbage creation) `BiConsumer<StringBuilder, Object>` emitter, which originally I defined in each class that had a 'stringafy' option. I ended up creating a utility function in the writer to make it easier, but it doesn't make much difference either way.

I agree that object->string formatting may be out of scope for the JsonWriter class, and could occur at a higher level.